### DISCUSSION TITLE
Better way to conditionally hide element based on select value

### DISCUSSION THREAD

**Discussion by LemonPie22** [filamentphp/filament#5168] [Help] [answered]
I have a select, `part_type_id`, as follows:

```
Forms\Components\Select::make('part_type_id')
    ->relationship('partType', 'name')
    ->label('Part type')
    ->required()
    ->autofocus()
    ->reactive()
    ->afterStateUpdated(
        function (Closure $set, $state) {
            $set(
                'dimension_unit_id',
                function (Closure $get) {
                    return PartType::where(
                        'id',
                        $get('part_type_id')
                    )
                        ->get(
                            'default_dim_unit_id'
                        )[0]['default_dim_unit_id'];
                }
            );
        }
    )
->default(1),
```

And I have a section, `component_items`, that I want to conditionally show and hide depending on the user's selection:

```
Forms\Components\Section::make('component_items')
    ->hidden(
        function (Closure $get) {
            return !PartType::where('id', $get('part_type_id'))
                ->get('has_items')[0]['has_items'];
        }
    )->label('Component items')
```

This works as long as I set a default value for `part_type_id` - something I'd rather not do. If I don't set this default I get `Undefined array key 0`, which makes sense. I guess I could put an `if` statement in my function but I suspect there may be a better way to do this. Any suggestions?

**Comment by joaovdiasb** [comment]
You could create `Hidden::make('has_items')` input to $set has_items inside afterStateUpdated or simply `PartType::where('id', $get('part_type_id'))->first()->has_items` or `PartType::where('id', $get('part_type_id'))->first()->default_dim_unit_id`

**Comment by danharrin** [comment, marked as answer]
```php
Forms\Components\Section::make('component_items')
    ->visible(fn (Closure $get) => PartType::find($get('part_type_id'), ['has_items'])?->has_items)
    ->label('Component items')
```

`find()` returns a PartType or null. Using the `?->` operator we can handle the null case in 1 line

**Reply by danharrin** [reply]
Also swapping `hidden()` for `visible()` means we dont need to invert the bool with `!`

**Reply by LemonPie22** [reply]
That works and I appreciate your advice to use `visible()` instead of `hidden()`. Makes my code simpler. Thanks @danharrin
